plugin.video.fanfilm/resources/lib/sources/__init__.py

```diff
@@ -0,0 +1,986 @@
+
+
+                    sysname, sysurl, sysimage, sysprovider = urllib.quote_plus(name), urllib.quote_plus(url), urllib.quote_plus(poster), urllib.quote_plus(provider)
+
+                    syssource = urllib.quote_plus(json.dumps([self.sources[i]]))
+                    if i == 0:
+                        query = 'action=playItem&content=%s&name=%s&year=%s&imdb=%s&tvdb=%s&source=%s&meta=%s' % (content, sysname, year, imdb, tvdb, syssource, sysmeta)
+                    else:
+                        query = 'action=playItem&content=%s&name=%s&year=%s&imdb=%s&tvdb=%s&source=%s' % (content, sysname, year, imdb, tvdb, syssource)
+
+                    cm = []
+                    cm.append((control.lang(30504).encode('utf-8'), 'RunPlugin(%s?action=queueItem)' % sysaddon))
+                    cm.append((control.lang(30505).encode('utf-8'), 'RunPlugin(%s?action=download&name=%s&image=%s&url=%s&provider=%s)' % (sysaddon, sysname, sysimage, sysurl, sysprovider)))
+                    cm.append((infoMenu, 'Action(Info)'))
+                    cm.append((control.lang(30506).encode('utf-8'), 'RunPlugin(%s?action=refresh)' % sysaddon))
+                    cm.append((control.lang(30507).encode('utf-8'), 'RunPlugin(%s?action=openSettings)' % sysaddon))
+                    cm.append((control.lang(30508).encode('utf-8'), 'RunPlugin(%s?action=openPlaylist)' % sysaddon))
+
+                    item = control.item(label=label, iconImage='DefaultVideo.png', thumbnailImage=thumb)
+                    try: item.setArt({'poster': poster, 'tvshow.poster': poster, 'season.poster': poster, 'banner': banner, 'tvshow.banner': banner, 'season.banner': banner})
+                    except: pass
+                    item.setInfo(type='Video', infoLabels = meta)
+                    if not fanart == None: item.setProperty('Fanart_Image', fanart)
+                    item.setProperty('Video', 'true')
+                    #item.setProperty('IsPlayable', 'true')
+                    item.addContextMenuItems(cm, replaceItems=True)
+
+                    control.addItem(handle=int(sys.argv[1]), url='%s?%s' % (sysaddon, query), listitem=item, isFolder=False)
+                except Exception as e:
+                    control.log('ERROR Sources.addItem %s' % e)
+
+                    pass
+            control.content(int(sys.argv[1]), 'files')
+            control.directory(int(sys.argv[1]), cacheToDisc=True)
+            try: self.progressDialog.close()
+            except: pass
+        except:
+            control.infoDialog(control.lang(30501).encode('utf-8'))
+            try: self.progressDialog.close()
+            except: pass
+
+
+    def playItem(self, content, name, year, imdb, tvdb, source):
+        control.resolve(int(sys.argv[1]), True, control.item(path=''))
+        control.execute('Dialog.Close(okdialog)')
+
+        next = [] ; prev = [] ; total = []
+        meta = None
+
+        for i in range(1,10000):
```

Comment:
To nie jest python3 jeśli się nie mylę. Zakres range(1,10000) już trochę za duży. Sugeruję xrange